src/os_posix/os_union_fs.c

```diff
@@ -891,27 +883,38 @@ __union_build_extents_from_dest_file_lseek(
 
     /* Check that we opened a valid file descriptor. */
     WT_ASSERT(session, fcntl(fd, F_GETFD) != -1 || errno != EBADF);
-    __union_fs_file_size((WT_FILE_HANDLE *)union_fh, (WT_SESSION *)session, &file_size);
+    WT_ERR(__union_fs_file_size((WT_FILE_HANDLE *)union_fh, (WT_SESSION *)session, &file_size));
```

Comment:
Update this function to take a WT_UNION_FILE_HANDLE instead I think. It immediately casts back.

Reply:
We'll need to leave this as `WT_FILE_HANDLE`. `__union_fs_file_size` is part of the file handle API called by WiredTiger.